File: Frontend/apoyo-management/src/app/services/corporate-lead.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environment/environment';

export interface CorporateLead {
  companyName: string;
  contactPerson: string;
  businessEmail: string;
  mobile: string;
  workforceCategory: string;
  estimatedStaffSize: number;
  location: string;
  projectDetails: string;
}

@Injectable({
  providedIn: 'root'
})
export class CorporateLeadService {

  private apiUrl = `${environment.apiUrl}/api/CorporateLeads`;

  constructor(private http: HttpClient) {}

  createLead(lead: CorporateLead): Observable<any> {
    return this.http.post(this.apiUrl, lead);
  }
}

File: Backend/ApoyoManagement.API/Models/CorporateLead.cs
namespace ApoyoManagement.API.Models;

public class CorporateLead
{
    public int Id { get; set; }

    public string CompanyName { get; set; } = string.Empty;

    public string ContactPerson { get; set; } = string.Empty;

    public string BusinessEmail { get; set; } = string.Empty;

    public string Mobile { get; set; } = string.Empty;

    public string WorkforceCategory { get; set; } = string.Empty;

    public int EstimatedStaffSize { get; set; }

    public string Location { get; set; } = string.Empty;

    public string ProjectDetails { get; set; } = string.Empty;

    // Lead Management
    public string Status { get; set; } = "New";

    public string? AssignedTo { get; set; }

    public DateTime? FollowUpDate { get; set; }

    public string? InternalNotes { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}